File: cloud-functions/submitTrade-js/exchanges/bingx/scripts/order.js
const CustomError = require('../../../utils/error.js');

async function getTpsStatus(session, symbol, tpSlOrders) {
    try {
        const fetchOrders = await session.currentOrders(symbol);
        const openOrders = fetchOrders["orders"];

        let tpsData = [];
        const activeStatus = ["NEW", "PARTIALLY_FILLED"];

        for (let tpOrder of tpSlOrders) {
            if ("tp_number" in tpOrder) {
                const tpOrderId = tpOrder["orderID"];
                const matchingOpenOrders = openOrders.filter(openOrder => String(openOrder["orderId"]) === tpOrderId);
                tpOrder["tp_status"] = matchingOpenOrders.length > 0 && activeStatus.includes(matchingOpenOrders[0]["status"]) ? "active" : "filled";
                tpsData.push(tpOrder);
            }
        }
        return tpsData;
    } catch(error) {
        throw new CustomError({
            message: `Error getting status of tps in scripts: ${error.message}`,
            status: 500,
            source: 'getTpsStatus',
        });
    }
}

module.exports = {getTpsStatus};

File: cloud-run/submitTrade/exchanges/bingx/scripts/order.py
async def get_tps_status(session, tp_orders, trade_info):
    symbol = trade_info["symbol"]

    fetch_orders = await session.current_orders(symbol=symbol)
    open_orders = fetch_orders["orders"]

    tps_data = []
    active_status = ["NEW", "PARTIALLY_FILLED"]

    for tp_order in tp_orders:
        tp_order_id = tp_order["orderID"]
        matching_open_orders = [open_order for open_order in open_orders if
                                open_order["orderId"] == tp_order_id]
        tp_order["tp_status"] = "active" if matching_open_orders and matching_open_orders[0][
            "status"] in active_status else "filled"
        tps_data.append(tp_order)
    return tps_data